crates/dwallet-classgroups-types/src/lib.rs

```diff
@@ -26,6 +26,51 @@ pub type SingleEncryptionKeyAndProof = (
 /// each prime corresponds to a dynamic object.
 pub const NUM_OF_CLASS_GROUPS_KEY_OBJECTS: usize = MAX_PRIMES;
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct RootSeed([u8; RootSeed::SEED_LENGTH]);
+
+impl RootSeed {
+    pub const SEED_LENGTH: usize = 32;
+
+    pub fn new(seed: [u8; Self::SEED_LENGTH]) -> Self {
+        RootSeed(seed)
+    }
+
+    /// Generates a cryptographically secure random seed.
+    pub fn random_seed() -> Self {
+        let mut bytes = [0u8; Self::SEED_LENGTH];
+        OsCsRng.fill_bytes(&mut bytes);
+        RootSeed(bytes)
+    }
+
+    /// Reads a class group seed (encoded in Base64) from a file.
+    pub fn from_file<P: AsRef<std::path::Path>>(path: P) -> DwalletMPCResult<Self> {
+        let contents = std::fs::read_to_string(path)
+            .map_err(|e| DwalletMPCError::FailedToReadCGKey(e.to_string()))?;
```

Comment:
rename the error to `FailedToReadClassGroupsKey`

Reply:
Better `FailedToReadRootSeed`